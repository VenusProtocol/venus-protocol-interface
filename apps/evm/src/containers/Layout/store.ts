import { create } from 'zustand';

import { createStoreSelectors } from 'utilities';

interface State {
  isScrollToTopVisible: boolean;
  setScrollToTopVisible: (isScrollToTopVisible: State['isScrollToTopVisible']) => void;
}

const useStore = create<State>()(set => ({
  isScrollToTopVisible: false,
  setScrollToTopVisible: (v: boolean) => set({ isScrollToTopVisible: v }),
}));

export const store = createStoreSelectors(useStore);
