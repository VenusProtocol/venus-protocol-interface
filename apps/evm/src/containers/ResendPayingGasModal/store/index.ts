import { create } from 'zustand';

import { createStoreSelectors } from 'utilities';
import type { StoreState } from '../types';

const useStore = create<StoreState<any>>()(set => ({
  lastFailedGaslessTransaction: undefined,
  openModal: ({ lastFailedGaslessTransaction }) => set({ lastFailedGaslessTransaction }),
  closeModal: () => set({ lastFailedGaslessTransaction: undefined }),
}));

export const store = createStoreSelectors(useStore);
