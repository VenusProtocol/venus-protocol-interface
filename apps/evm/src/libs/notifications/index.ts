export * from './NotificationCenter';
export * from './utilities';
export * from './types';
