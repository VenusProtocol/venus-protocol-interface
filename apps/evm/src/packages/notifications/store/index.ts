import { create } from 'zustand';

import { createStoreSelectors } from 'utilities/createStoreSelectors';

import { Notification, StoreState } from './types';

const useStore = create<StoreState>()(set => ({
  notifications: [],

  addNotification: input => {
    const id = input.id ?? Date.now();

    set(state => {
      const notification: Notification = {
        ...input,
        id,
        onClose: () =>
          state.removeNotification({
            id,
          }),
      };

      return {
        notifications: [notification, ...state.notifications],
      };
    });

    return id;
  },

  updateNotification: input =>
    set(state => ({
      notifications: state.notifications.map(notification =>
        notification.id === input.id ? { ...notification, ...input } : notification,
      ),
    })),

  removeNotification: ({ id }) =>
    set(state => ({
      notifications: state.notifications.filter(notification => notification.id !== id),
    })),
}));

export const store = createStoreSelectors(useStore);
