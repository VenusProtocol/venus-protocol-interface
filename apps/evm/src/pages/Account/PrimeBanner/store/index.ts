import createDeepMerge from '@fastify/deepmerge';
import { createStoreSelectors } from 'utilities';
import { create } from 'zustand';
import { persist } from 'zustand/middleware';

export interface State {
  doNotShowPrimePromotionalBanner: boolean;
  hidePrimePromotionalBanner: () => void;
}

const deepMerge = createDeepMerge({ all: true });

const useStore = create<State>()(
  persist(
    set => ({
      doNotShowPrimePromotionalBanner: false,
      hidePrimePromotionalBanner: () =>
        set(state => ({
          ...state,
          doNotShowPrimePromotionalBanner: true,
        })),
    }),
    {
      name: 'venus-prime-banner-store',
      merge: (persisted, current) => deepMerge(current, persisted) as never,
    },
  ),
);

export const store = createStoreSelectors(useStore);
