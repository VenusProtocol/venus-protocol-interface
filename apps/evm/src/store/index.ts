import createDeepMerge from '@fastify/deepmerge';
import { ChainId } from 'types';
import { createStoreSelectors, extractEnumValues } from 'utilities';
import { create } from 'zustand';
import { persist } from 'zustand/middleware';
import { immer } from 'zustand/middleware/immer';

export interface UserChainSettings {
  gaslessTransactions: boolean;
  showPausedAssets: boolean;
  showUserAssetsOnly: boolean;
}

type UserSettings = Partial<Record<ChainId, Partial<UserChainSettings>>>;

export interface State {
  userSettings: UserSettings;
  setUserSettings: (input: {
    settings: Partial<UserChainSettings>;
    chainIds?: ChainId[];
  }) => void;
}

const deepMerge = createDeepMerge({ all: true });

const allChainIds = extractEnumValues(ChainId);
export const initialUserSettings: UserSettings = {
  [ChainId.ZKSYNC_MAINNET]: {
    gaslessTransactions: true,
  },
  [ChainId.ZKSYNC_SEPOLIA]: {
    gaslessTransactions: true,
  },
};

const useStore = create<State>()(
  persist(
    immer(set => ({
      userSettings: initialUserSettings,
      setUserSettings: ({ settings, chainIds = allChainIds }) =>
        set(state =>
          chainIds.forEach(chainId =>
            Object.entries(settings).forEach(([key, value]) => {
              if (!state.userSettings[chainId]) {
                state.userSettings[chainId] = {};
              }

              state.userSettings[chainId][key as keyof UserChainSettings] = value;
            }),
          ),
        ),
    })),
    {
      name: 'venus-global-store',
      merge: (persisted, current) => deepMerge(current, persisted) as never,
    },
  ),
);

export const store = createStoreSelectors(useStore);
