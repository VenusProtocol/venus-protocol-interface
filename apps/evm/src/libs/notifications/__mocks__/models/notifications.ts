import { Notification } from '../../store/types';

export const notifications: Notification[] = [
  {
    id: 0,
    description: 'Fake description 0',
    variant: 'success',
  },
  {
    id: 1,
    title: 'Fake title 1',
    description: 'Fake description 1',
  },
];
