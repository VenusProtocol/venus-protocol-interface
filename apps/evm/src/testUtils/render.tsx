import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render as renderComponentTl } from '@testing-library/react';
import { renderHook as renderHookTl } from '@testing-library/react-hooks';
import type { ReactElement } from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import type Vi from 'vitest';

import fakeSigner from '__mocks__/models/signer';

import { Web3Wrapper, useAccountAddress, useChainId, useSigner } from 'libs/wallet';
import { MuiThemeProvider } from 'theme/MuiThemeProvider';
import type { ChainId } from 'types';

const createQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        gcTime: 0,
      },
    },
  });

interface Options {
  accountAddress?: string;
  chainId?: ChainId;
  routerInitialEntries?: string[];
  routePath?: string;
}

interface WrapperProps {
  queryClient: QueryClient;
  children?: React.ReactNode;
  options?: Partial<Options>;
}

const Wrapper: React.FC<WrapperProps> = ({ children, queryClient, options }) => {
  if (options?.accountAddress) {
    const accountAddress = options?.accountAddress;

    (useAccountAddress as Vi.Mock).mockImplementation(() => ({
      accountAddress,
    }));

    (useSigner as Vi.Mock).mockImplementation(() => ({
      signer: {
        ...fakeSigner,
        getAddress: async () => accountAddress,
      },
    }));
  }

  if (options?.chainId) {
    (useChainId as Vi.Mock).mockImplementation(() => ({
      chainId: options?.chainId,
    }));
  }

  return (
    <MuiThemeProvider>
      <QueryClientProvider client={queryClient}>
        <Web3Wrapper>
          <MemoryRouter initialEntries={options?.routerInitialEntries || ['/']}>
            <Routes>
              <Route path={options?.routePath || '/'} element={children} />
            </Routes>
          </MemoryRouter>
        </Web3Wrapper>
      </QueryClientProvider>
    </MuiThemeProvider>
  );
};

export const renderComponent = (children: ReactElement, options?: Partial<Options>) => {
  const queryClient = createQueryClient();

  const renderRes = renderComponentTl(children, {
    wrapper: props => <Wrapper queryClient={queryClient} options={options} {...props} />,
  });

  return {
    ...renderRes,
    queryClient,
  };
};

export const renderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  options?: Partial<Options>,
) => {
  const queryClient = createQueryClient();

  const renderRes = renderHookTl(hook, {
    wrapper: props => <Wrapper queryClient={queryClient} options={options} {...props} />,
  });

  return {
    ...renderRes,
    queryClient,
  };
};
