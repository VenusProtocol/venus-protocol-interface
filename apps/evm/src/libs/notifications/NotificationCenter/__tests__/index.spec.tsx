import { renderComponent } from 'testUtils/render';

import { store } from 'libs/notifications/store';

import NotificationCenter from '..';
import { notifications as fakeNotifications } from '../../__mocks__/models/notifications';
import TEST_IDS from '../testIds';

describe('NotificationCenter', () => {
  it('renders without crashing', () => {
    renderComponent(<NotificationCenter />);
  });

  it('renders notifications correctly', () => {
    // Add fake notifications to the store
    store.setState({
      notifications: fakeNotifications,
    });

    const { getByTestId } = renderComponent(<NotificationCenter />);

    expect(getByTestId(TEST_IDS.container).textContent).toMatchSnapshot();
  });
});
