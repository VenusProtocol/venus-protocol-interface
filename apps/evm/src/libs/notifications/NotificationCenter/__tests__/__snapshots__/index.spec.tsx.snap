// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotificationCenter > renders notifications correctly 1`] = `"Fake description 0Fake title 1Fake description 1"`;
